src/platform/storm/lib.rs

```diff
@@ -68,17 +69,18 @@ impl Firestorm {
             4 => f(Some(self.spi)),
             5 => f(Some(self.nrf51822)),
             6 => f(Some(self.isl29035)),
-            _ => f(None)
+            _ => f(None),
         }
     }
 }
 
 macro_rules! static_init {
-   ($V:ident : $T:ty = $e:expr) => {
-        let $V : &mut $T = {
+   ($V:ident : $T:ty = $e:expr, $size:expr) => {
+        let $V : &'static mut $T = {
             // Waiting out for size_of to be available at compile-time to avoid
             // hardcoding an abitrary large size...
-            static mut BUF : [u8; 1024] = [0; 1024];
+            static mut BUF : [u8; $size] = [0; $size];
+            assert!(mem::size_of::<$T>() <= $size);
```

Comment:
if we're OK with tight bounds, it's actually possible to get the compiler to verify this statically. Something like:

```rust
let _ : T = mem::transmute(BUF);
```